### PR TITLE
Suggested fix for issue #59 - Serial tool window exception

### PR DESCRIPTION
**Pull request by Basirk**
Used this commit from another plugin to suggest a fix for this one:
https://github.com/antlr/intellij-plugin-v4/commit/097170b095231013103a2024d9237e953ce61c1c
Note - untested and Java is not my main language.

### CODE REVIEW

**Comment by eiryu** [comment]
#59

File: src/com/vladsch/clionarduinoplugin/serial/SerialMonitorToolWindow.java [hunk 1]
```diff
@@ -47,20 +47,22 @@ public class SerialMonitorToolWindow {
...
-        SimpleToolWindowPanel mainPanel = new SimpleToolWindowPanel(false, true);
-
-        mySerialMonitorPanel = new SerialMonitorPanel(project);
-        mainPanel.add(mySerialMonitorPanel.getComponent());
-
-        ContentFactory contentFactory = ContentFactory.SERVICE.getInstance();
-        Content content = contentFactory.createContent(mainPanel, Bundle.message("toolwindow.serial-monitor.title"), false);
-        myToolWindowContentMap.put(SERIAL_MONITOR_ID, content);
-        toolWindow.getContentManager().addContent(content);
+        toolWindowManager.invokeLater(() -> {
```
Comment (eiryu) [inline src/com/vladsch/clionarduinoplugin/serial/SerialMonitorToolWindow.java:50]: I changed it this way and it worked.

```suggestion
        ToolWindowManager.getInstance(project).invokeLater(() -> {
```

https://github.com/eiryu/CLionArduinoPlugin/pull/1/commits/a7c5854b4fd077cc39d8839d2c318da4f31e5ac3

Comment (eiryu) [reply src/com/vladsch/clionarduinoplugin/serial/SerialMonitorToolWindow.java:50]: image

File: src/com/vladsch/clionarduinoplugin/serial/SerialMonitorToolWindow.java [hunk 2]
```diff
@@ -47,20 +47,22 @@ public class SerialMonitorToolWindow {
...
+            toolWindow = ToolWindowManager.getInstance(project).registerToolWindow(TOOL_WINDOW_ID, false, ToolWindowAnchor.BOTTOM, project, true);
```
Comment (eiryu) [inline src/com/vladsch/clionarduinoplugin/serial/SerialMonitorToolWindow.java:51]: introduce local variable

```suggestion
        ToolWindowManager toolWindowManager = ToolWindowManager.getInstance(project);
        toolWindowManager.invokeLater(() -> {
            toolWindow = toolWindowManager.registerToolWindow(TOOL_WINDOW_ID, false, ToolWindowAnchor.BOTTOM, project, true);
```

Comment (goodtho) [reply src/com/vladsch/clionarduinoplugin/serial/SerialMonitorToolWindow.java:51]: Thank you to come up with this fix!
Uhm.. Where/how can I change an installed CLion Plugin? I can just find the CLionArduinoPlugin binary File..

Comment (eiryu) [reply src/com/vladsch/clionarduinoplugin/serial/SerialMonitorToolWindow.java:51]: If you create a zip or jar, you may be able to install it. (I have not tried it.)
https://www.jetbrains.com/help/idea/managing-plugins.html#install_plugin_from_disk

Comment (JensvandeWiel) [reply src/com/vladsch/clionarduinoplugin/serial/SerialMonitorToolWindow.java:51]: why is this not added 😠

**Comment by TCB13** [comment]
Please merge and deploy this.

File: src/com/vladsch/clionarduinoplugin/serial/SerialMonitorToolWindow.java (hunk 2 above)
Comment (TCB13) [reply src/com/vladsch/clionarduinoplugin/serial/SerialMonitorToolWindow.java:51]: > If you create a zip or jar, you may be able to install it. (I have not tried it.) https://www.jetbrains.com/help/idea/managing-plugins.html#install_plugin_from_disk

Gives me: Fail to load plugin descriptor from file CLionArduinoPlugin-patch-1.zip

**Comment by vsch** [comment]
New version of plugin has this issue fixed. JetBrains should have it approved soon, but there is a direct link to this version before approval:  https://plugins.jetbrains.com/plugin/11301-arduino-support/versions/stable/305519.

Works with CLion up to 2022.2. 2022.3 is broken and it will take some work to get it working with it.

**Pull request closed without merging**